### DISCUSSION TITLE
Using teller as an SDK

### DISCUSSION THREAD

**Discussion by marcelmindemann** [tellerops/teller#172] [General]
Hi,
is it possible or even encouraged to re-use `tellerops/teller/pkg` in our own code? We use an inhouse solution to frequently refresh and rotate secrets in the AWS parameter store, and we would like to programmatically distribute these secrets to various sinks. 
Instead of implementing the API of all possible sinks ourselves, it would be logical to re-use the implementations of providers that teller already has. However, as the documentation only talks about the CLI use case of teller, I wonder if that's a sustainable solution. For example, are you following semantic versioning practices also in the interfaces of your code, or only in the interface of your command line?

**Comment by jondot** [comment]
Yes it was built for this use case, feel free. 
if you need any help feel free to ping me
